### PR TITLE
Update defaults

### LINKED ISSUE #1329: Meaningless error values on Theory and Normalised Residuals
There is a long standing issue, from at least as far back as Sasview 3

The error values associated with both Theory and Normalised Residual workspaces are all set to 1.0  (This is the dY column in "show info" on a workspace array.)

It happens that the statistical error on the Normalised residual for the default "Use dI" weighting scheme should actually be 1.0 (see below), but I suspect that all these errors are simply being set to 1.0 by default?

Discussion needed please by SAS experts as to what we should do here.

Regarding the Theory workspace - we could attempt to propagate error estimates from the errors on the fit parameters. We could simply not store and save any error value at all.

Regarding the Normalised Residuals - We minimise chi2=sum{ ( (I-C)/W )^2 }, the "normalised residual" is R = (I-C)/W where I is data, C is calculated, and W the weight for each data point.

Default is weight W = dI the error on data I, thus R = (I+-dI)/dI - C/dI, so error on normalised residual, dR = 1

If W=1 (the "none" option), dR = dI

If W=sqrt(I) then dR = dI /sqrt(I),  with the usual headache if I=0

If W =I then dR = dI/I,  with the usual headache if I=0

### PR DESCRIPTION
**Pull request by butlerpd**
## Description
This pull request updates a number of defaults that have raised complaints and in the process fixed a few bugs found along the way.
* The number of points in the slicer now default to 100 instead of 20 or 36 (depending on slicer)
  * In the process, the code was refactored slightly as there were several places where the default was hard coded which overrode each other in unexpected ways - all now have a single spot for the default value
  * Also the box slicer did not have a validation method implemented causing it to fail if the user entered 0 points for example (the validation method on the other slicers already prevented that) This was also fixed
* The number of point in the theory curve was upped from 50 to 150 and moved to log spacing by default vs linear.
  * Again, the point spacing default setting was not being respected and the code was refactored with input from @rozyczko  to clean iup the whole process of setting that value.
* The polydispersity plots now default to linear linear rather than log log.
* The residual now no longer have error bars associated. They were originally set to 1 which is meaningless. In fact the scatter of those points is what matters and 95% of them should be between -3 and +3 (i.e withing 3 sigma)
  * The setting of the y axis to linear was found to be done in 2 places again leading to potentially unexpected results. It was left in the definition of the residuals plot object and removed from the plot residuals method
    * Note this could interact with the changes made by @caitwolf to the default axis for residuals in the case of sesans. It looks like it is in fact OK. 
  * Ideally there should be a line (or dashed line) on the residuals plot showing the -3 and +3 but that is currently not easy to do (according to @rozyczko)

I do not believe there is any documentation that is affected by these changes.

Fixes #1329,  

## How Has This Been Tested?

Every step was tested on Windows Server 2022 Datacenter, version 21H2 (essentially widows 10).  Tests were made using different default values, and different paths to things were often tested to verify there were no surprises --> this led to the bug fixes mentioned above.

## Review Checklist (please remove items if they don't apply):

- [x] Code has been reviewed
- [x] Functionality has been tested
- [ ] Windows installer (GH artifact) has been tested (installed and worked) 
- [ ] MacOSX installer (GH artifact) has been tested (installed and worked) 
- [x] The introduced changes comply with SasView license (BSD 3-Clause)

### CODE REVIEW

File: src/sas/qtgui/Plotting/Slicers/AnnulusSlicer.py
Comment (rozyczko) [inline src/sas/qtgui/Plotting/Slicers/AnnulusSlicer.py:111]: Is this commented section supposed to remain commented out? What happens if the `nbins` argument is None (which is the default)?

**Review by rozyczko — APPROVED** [review]
Commented on a minor issue - otherwise looks and behaves fine.
One issue is performance hit for real-time updates to the "log scale" checkbox and the number of points. This, however, is expected.

**Review by krzywon — APPROVED** [review]
Does what it says it does. I left a couple of nitpicky suggestions, but nothing that should hold this up.

File: src/sas/qtgui/Plotting/Slicers/AnnulusSlicer.py
Comment (butlerpd) [reply src/sas/qtgui/Plotting/Slicers/AnnulusSlicer.py:111]: woops ... good catch. Actually the comments just above the section are also wrong now. I will remove all those comments. 

Regarding what happens if `nbins = None`, it will be reset to the default value set at the beginning by the validate method.  That was not the case for the box slicer but it is now :-)

File: src/sas/qtgui/Plotting/Slicers/AnnulusSlicer.py
Comment (butlerpd) [reply src/sas/qtgui/Plotting/Slicers/AnnulusSlicer.py:31]: Beyond my skill level at this point but if it would make the code better/cleaner for the future happy to be educated .. or have someone else do it properly.  I assume we can merge this and then discuss if it is worth doing the above and if so how?

File: src/sas/qtgui/Perspectives/Fitting/FittingUtilities.py
Comment (butlerpd) [reply src/sas/qtgui/Perspectives/Fitting/FittingUtilities.py:545]: I have no idea but there are a lot of instances of Data1D.x, .dx, .etc being set to None so just doing a "monkey see/monkey do."  If it would be better coding practice to not state ... and it really does default to None that would be fine by me.

**Review by caitwolf — APPROVED** [review]
It is behaving as it should and this did not break any of the changes I made on the SESANS residuals plots that was mentioned above.

File: src/sas/qtgui/Perspectives/Fitting/FittingUtilities.py
Comment (krzywon) [reply src/sas/qtgui/Perspectives/Fitting/FittingUtilities.py:545]: An explicit setting of something doesn't hurt, I was just wondering if there was a specific reason to do so in this case. Leaving it set to None will ensure the values are always what they are meant to be.

**Pull request merged**
